### PR TITLE
move GeoJSON creation to `api.py` and add `as_dict` method

### LINKED ISSUE #607: Remove `api/test_indicator_geojson_io.py` in favor of `api/test_indicators.py`
- [ ] make sure to test `test_invalid_set_of_arguments(…)` for both, JSON and GeoJSON

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description
- feat(indicator): add as_dict method
- refactor(oqt/api): move geojson creation to api

### Corresponding issue
Closes #607 #636 #655

### New or changed dependencies
- None

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

File: ohsome_quality_analyst/api/request_models.py [hunk 1]
```diff
@@ -29,7 +50,7 @@ class IndicatorRequest(BaseBpolys):
         ...,
         title="Topic Key",
         alias="topic",
-        example="building-count",
+        # example="building-count",
```
Comment (joker234) [inline ohsome_quality_analyst/api/request_models.py:53]: Why is this commented out and not removed? If you need it later again, you could add a TODO.

**Review by joker234 — COMMENTED** [review]
Why do we need the .joblib files in here? Isn't the machine learning library an external packages which is supposed to be tested there?

File: tests/integrationtests/test_geodatabase.py [hunk 2]
```diff
@@ -1,73 +1,67 @@
...
+    assert len(result) == 1
+
+
+def test_get_shdi_single_intersection_featurecollection(
+    feature_collection_germany_heidelberg_bahnstadt_bergheim,
+):
+    result = asyncio.run(
+        db_client.get_shdi(feature_collection_germany_heidelberg_bahnstadt_bergheim)
+    )
+    assert len(result) == 3
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integrationtests/test_geodatabase.py:35]: The fixture contains three features.

File: ohsome_quality_analyst/api/request_models.py (hunk 1 above)
Comment (matthiasschaub) [reply ohsome_quality_analyst/api/request_models.py:53]:
```suggestion
```

File: tests/integrationtests/test_oqt.py [hunk 3]
```diff
@@ -25,7 +18,28 @@
...
+@pytest.mark.parametrize(
+    "indicator,topic",
+    [
+        ("minimal", "topic_minimal"),
+        ("mapping-saturation", "topic_building_count"),
+        ("currentness", "topic_building_count"),
+        ("attribute-completeness", "topic_building_count"),
+    ],
+)
+def test__create_indicator(feature, indicator, topic, request):
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integrationtests/test_oqt.py:42]: Done

File: tests/integrationtests/test_oqt.py [hunk 4]
```diff
@@ -36,74 +50,49 @@ def test_create_indicator(feature, indicator, topic, request):
 
 
 @oqt_vcr.use_cassette
-def test_create_report(feature):
+def test__create_report(feature):
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integrationtests/test_oqt.py:53]: Done

File: tests/integrationtests/test_oqt.py [hunk 5]
```diff
@@ -36,74 +50,38 @@ def test_create_indicator(feature, indicator, topic, request):
...
-                },
-            },
-        )
-        asyncio.run(oqt.create_indicator(parameters, key="mapping-saturation"))
-
-
-if __name__ == "__main__":
-    unittest.main()
+@mock.patch.dict("os.environ", {"OQT_GEOM_SIZE_LIMIT": "1"}, clear=True)
+def test_create_indicator_size_limit_bpolys(
```
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integrationtests/test_oqt.py:63]: seperated them again

File: ohsome_quality_analyst/api/request_models.py (hunk 1 above)
Comment (matthiasschaub) [reply ohsome_quality_analyst/api/request_models.py:53]: removed comment

File: tests/integrationtests/test_geodatabase.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (joker234) [reply tests/integrationtests/test_geodatabase.py:35]: Please check again, as `heidelberg-bahnstadt-bergheim-featurecollection.geojson` has a similar name as `feature_collection_germany_heidelberg_bahnstadt_bergheim`

File: tests/integrationtests/test_oqt.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (joker234) [reply tests/integrationtests/test_oqt.py]: thanks. Please consider testing `oqt.create_indicator(…)` more thoroughly (i.e. with FeatureCollections).

**Review by joker234 — COMMENTED** [review]
see comments

File: tests/integrationtests/test_geodatabase.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integrationtests/test_geodatabase.py:35]: renamed

[the remark this answers is not in the document]
Comment (matthiasschaub) [reply tests/integrationtests/test_geodatabase.py:35]: and add another test case

**Pull request merged**